src/internet_identity/src/assets.rs

```diff
@@ -19,15 +21,43 @@ pub enum ContentType {
     SVG
 }
 
-pub fn for_each_asset(mut f: impl FnMut(&'static str, ContentEncoding, ContentType, &'static [u8], &[u8; 32])) {
+lazy_static! {
+    // The <script> tag that sets the canister ID and loads the 'index.js'
+    static ref INDEX_HTML_SETUP_JS: String = {
+        let canister_id = api::id();
+        format!(r#"var canisterId = '{canister_id}';let s = document.createElement('script');s.async = false;s.src = 'index.js';document.head.appendChild(s);"#)
+    };
 
-    let index_html = include_bytes!("../../../dist/index.html");
+    // The SRI sha256 hash of the script tag, used by the CSP policy.
+    // https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/script-src
+    pub static ref INDEX_HTML_SETUP_JS_SRI_HASH: String = {
+        let hash = &sha2::Sha256::digest(INDEX_HTML_SETUP_JS.as_bytes());
+        let hash = base64::encode(hash);
+        format!("sha256-{hash}")
+    };
 
-    let assets: [ (&str, &[u8], ContentEncoding, ContentType); 8] = [
+    // The full content of the index.html, after the canister ID (and script tag) have been
+    // injected
+    static ref INDEX_HTML_STR: String = {
+        let index_html = include_str!("../../../dist/index.html");
+        let foo: String = INDEX_HTML_SETUP_JS.to_string();
+        let index_html = index_html.replace(
+            "<script id='setupJs'></script>",
+            &format!("<script id='setupJs'>{foo}</script>").to_string()
+        );
+        index_html
+    };
+}
+
+// Get all the assets. Duplicated assets like index.html are shared and generally all assets are
+// prepared only once (like injecting the canister ID).
+pub fn get_assets() -> [ (&'static str, &'static [u8], ContentEncoding, ContentType); 8]  {
+    let index_html: &[u8] = INDEX_HTML_STR.as_bytes();
+    [
          ("/",
-            index_html,
-            ContentEncoding::Identity,
-            ContentType::HTML,
+             index_html,
+             ContentEncoding::Identity,
+             ContentType::HTML,
```

Comment:
Is a formatter run as part of CI?  I would not expect to see formatting-only changes in a PR.

Reply:
I didn't even remember we did, so I created this: https://dfinity.atlassian.net/browse/L2-230

@bitdivine no, the first item was off so I aligned it, didn't realize it wasn't like that before. Happy to revert if you feel strongly about it